scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -66,22 +70,71 @@ case object RedundantBraces extends Rewrite {
                 builder += TokenPatch.Remove(closeBrace)
               case _ =>
             }
+          case _ =>
         }
-      case d: Defn.Def if isCandidate(d, ctx) =>
+
+      case d: Defn.Def if isDefnCandidate(d) =>
         val open = d.body.tokens.head
         val close = d.body.tokens.last
         val bodyStatement = d.body match {
           case t: Term.Block => t.stats.head
           case _ => d.body
         }
-        val next = nextToken(close)
-        if (next.is[LF] &&
-          close.pos.start.line != bodyStatement.pos.end.line)
-          builder += TokenPatch.Remove(next)
-
+        removeTrailingLF(bodyStatement.pos, close)
         builder += TokenPatch.Remove(open)
         builder += TokenPatch.Remove(close)
+
+      case x: Term.If if settings.ifElseClauses =>
```

Comment:
I'm wondering if we can generalize the implementation to handle all single statement Term.Blocks in one go. What happens if we do something like the following psuedo code? 
```scala
case b: Term.Block if b.stats.lengthCompare(1) == 0 =>
  for {
    open <- b.tokens.headOption
    if open.is[Token.LeftBrace]
    close <- b.tokens.headOption
    if close.is[Token.RightBrace]
  } yield List(remove(open), remove(left), removeTrailingLF(...))
```

That could handle if + else + case clauses. If there's a wish for more detailed configuration  we could do it against `b.parent.get.productPrefix` which returns for example "Term.If" or "Case"

Reply:
That appeals to me much more conceptually but why wasn't it done that way before? The fact that it was only done for a few small cases suggests to me that a holistic approach didn't work... Is that not so? I'd be interested to know the history.

Reply:
Bah, I'm going down this path regardless. It's looking very promising! :D